primitives/core/src/ecdsa.rs

```diff
@@ -379,12 +379,12 @@ impl From<(secp256k1::Signature, secp256k1::RecoveryId)> for Signature {
 }
 
 #[cfg(feature = "full_crypto")]
-impl<'a> TryFrom<&'a Signature> for (secp256k1::Signature, secp256k1::RecoveryId) {
+impl<'a> TryFrom<&'a Signature> for (libsecp256k1::Signature, libsecp256k1::RecoveryId) {
 	type Error = ();
-	fn try_from(x: &'a Signature) -> Result<(secp256k1::Signature, secp256k1::RecoveryId), Self::Error> {
+	fn try_from(x: &'a Signature) -> Result<(libsecp256k1::Signature, libsecp256k1::RecoveryId), Self::Error> {
 		Ok((
-			secp256k1::Signature::parse_slice(&x.0[0..64]).expect("hardcoded to 64 bytes; qed"),
-			secp256k1::RecoveryId::parse(x.0[64]).map_err(|_| ())?,
+			libsecp256k1::Signature::parse_standard_slice(&x.0[0..64]).expect("hardcoded to 64 bytes; qed"),
```

Comment:
```suggestion
			libsecp256k1::Signature::parse_overflowing_slice(&x.0[0..64]).expect("hardcoded to 64 bytes; qed"),
```

We still need this as `parse_overflowing_slice` for now, then move to `parse_standard_slice` through a migration.

Reply:
What's the reasoning behind this? (just for my own education)

Reply:
Actually, I see. Some preexisting Signatures may be overflowing so we need to parse them with that

Reply:
It would otherwise be an actual security advisory for Substrate.

Parsing overflowing slice is not dangerous and by itself is safe. The security advisory arises if there are multiple clients being maintained and all of them need to conform to a standard way.

If we change this to `parse_standard_slice` in Substrate now we need to mark it as a high security upgrade for Substrate and ask everyone to upgrade, and during this period an attacker can split the chain. On the other hand, if we do it via a migration then it's completely safe, and we can do the migration in our own pace without any security concerns.

The drawback for the later, of course, is that `parse_overflowing_slice` will exist in Substrate at least for years to come (until when old blocks no longer needs to be re-processed by anyone). However, given that the method is actually safe, but just non-standard, it shouldn't be of any problems.

Reply:
What you mean by a migration? 

Isn't the problem only the old runtimes? So we should introduce a new host function?

Reply:
Yeah. We're on the same page.